subsys/bluetooth/host/hci_core.c

```diff
@@ -3088,6 +3092,9 @@ static void hci_reset_complete(struct net_buf *buf)
 	}
 
 	scan_dev_found_cb = NULL;
+#if defined(CONFIG_BT_HCI_VENDOR_EVENT_CALLBACK)
+	hci_vnd_evt_cb = NULL;
+#endif /* CONFIG_BT_HCI_VENDOR_EVENT_CALLBACK */
```

Comment:
This doesn't seem to be the right place to do this. The other callbacks that are set to NULL here are related to active LE roles which get implicitly disable incase HCI_Reset is sent, however a vendor event callback should be fine to have available at any point, including getting registered before initialising the reset of the stack. Since the variable is declared static I guess you don't need this explicit setting to NULL anywhere.

Reply:
Yes, good point